cocotb/monitors/avalon.py

```diff
@@ -222,20 +230,3 @@ def valid():
                             raise AvalonProtocolError(
                                 "In-Packet Timeout. Didn't receive any valid data for %d cycles!" %
                                 invalid_cyclecount)
-
-class AvalonSTPktsWithChannel(AvalonSTPkts):
-    """Packetized AvalonST bus using channel."""
-
-    _signals = ["valid", "data", "startofpacket", "endofpacket", "channel"]
-    _optional_signals = ["error", "ready", "empty"]
-
-    def __init__(self, *args, **kwargs):
-        AvalonSTPkts.__init__(self, *args, **kwargs)
-
-    def _recv(self, pkt):
-        """Force use of channel in recv function.
-
-        Args:
-            pkt: (string) Monitored data.
-        """
-        AvalonSTPkts._recv(self, {"data": pkt, "channel": self.channel})
```

Comment:
Again, removing this class will break any existing code that uses it.

It looks like you could keep old code working by removing everything but `__init__`, and changing it to pass `record_channel=True` to `__init__`? Perhaps:
```
def __init__(*args, **kwargs):
    warnings.warn(
        "AvalonSTPktsWithChannel is deprecated, use AvalonSTPkts(..., record_channel=True)",
        DeprecationWarning, stacklevel=2)
    AvalonSTPkts.__init__(self, *args, record_channel=True, **kwargs)

Reply:
Can I use [deprecated](https://pypi.org/project/Deprecated/) ? I really think we should get rid of this old legacy somewhere in the future.

Reply:
I'd prefer not to in this patch. Cocotb currently has no dependencies, we should have that discussion independently.

Perhaps open an issue suggesting we replace all instances of `DeprecationWarning` with that third party module, and keep doing it by hand for now.

Agreed that it's nice to remove legacy, but if it's user facing we should always deprecate first.

Reply:
warn & DeprecationWarning is fine. Was just used to @deprecated.